pkg/build/builder/cmd/builder.go

```diff
@@ -45,9 +45,13 @@ func run(builderFactory factoryFunc) {
 			glog.Fatalf("Cannot parse build URL: %s", build.Spec.Source.Git.URI)
 		}
 		scmAuths := auths(sourceURL)
-		if err := setupSourceSecret(build.Spec.Source.SourceSecret.Name, scmAuths); err != nil {
+		url, err := setupSourceSecret(build.Spec.Source.SourceSecret.Name, scmAuths)
```

Comment:
alternately, pass in the `build.Spec.Source.Git.URI` and assign the output, so each auth gets a crack at mutating it as appropriate